### PR TITLE
Remove unsupported tests (ADFS v2 v3, iOS, Android)

### PR DESCRIPTION
**Pull request by bgavrilMS**
ADFSv2 and ADFSv3 are now out of support.

### CODE REVIEW

**Comment by bgavrilMS** [comment]
@trwalke - please review the changes to the public API

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OnBehalfOfTests.cs [hunk 1]
```diff
@@ -220,11 +220,11 @@ public async Task OboAndClientCredentials_WithRegional_ReturnsCorrectTokens_Test
             Assert.AreEqual(TokenSource.Cache, clientResult.AuthenticationResultMetadata.TokenSource);
         }
 
-        [RunOn(TargetFrameworks.NetCore)]
+        [RunOn(TargetFrameworks.NetFx)]
         public async Task WithMultipleUsers_TestAsync()
         {
             var aadUser1 = (await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false)).User;
-            var aadUser2 = (await LabUserHelper.GetAdfsUserAsync(FederationProvider.AdfsV2, true).ConfigureAwait(false)).User;
+            var aadUser2 = (await LabUserHelper.GetDefaultUser2Async().ConfigureAwait(false)).User;
```
Comment (gladjohn) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OnBehalfOfTests.cs:227]: ah, we never needed an ADFS user for this test. We should make it a point to always rely on cloud users for our tests unless if it is ADFS specific.

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs [hunk 2]
```diff
@@ -69,8 +69,8 @@ public async Task WithCertificate_TestAsync(Cloud cloud, TargetFrameworks runOn,
         }
 
         [TestMethod]
-        [DataRow(Cloud.Public, TargetFrameworks.NetFx | TargetFrameworks.NetCore)]
-        [DataRow(Cloud.Adfs, TargetFrameworks.NetFx)]
+        //[DataRow(Cloud.Public, TargetFrameworks.NetFx | TargetFrameworks.NetCore)]
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ClientCredentialsTests.cs:72]:
```suggestion
        [DataRow(Cloud.Public, TargetFrameworks.NetFx | TargetFrameworks.NetCore)]
```

File: tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OnBehalfOfTests.cs [hunk 3]
```diff
@@ -429,8 +429,12 @@ void AssertLastHttpContent(string content)
             catch (MsalUiRequiredException)
             {
                 Assert.IsFalse(silentCallShouldSucceed, "ATS should have found a token, but it didn't");
+                string u = user.Upn;
+                string p = user.GetOrFetchPassword();
+
                 authResult = await pca
                     .AcquireTokenByUsernamePassword(s_oboServiceScope, user.Upn, user.GetOrFetchPassword())
+                    //.AcquireTokenInteractive(s_oboServiceScope)
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/OnBehalfOfTests.cs:437]:
```suggestion
                Assert.IsFalse(silentCallShouldSucceed, "ATS should have found a token, but it didn't");
                authResult = await pca
                    .AcquireTokenByUsernamePassword(s_oboServiceScope, user.Upn, user.GetOrFetchPassword())
```

File: tests/Microsoft.Identity.Test.Integration.netfx/Infrastructure/SeleniumExtensions.cs [hunk 4]
```diff
@@ -29,7 +29,7 @@ public static IWebDriver CreateDefaultWebDriver()
 
             // ~2x faster, no visual rendering
             // remove when debugging to see the UI automation
-            options.AddArguments("headless");
+            //options.AddArguments("headless");
```
Comment (pmaytak) [inline tests/Microsoft.Identity.Test.Integration.netfx/Infrastructure/SeleniumExtensions.cs:32]:
```suggestion
            options.AddArguments("headless");
```

**Review by pmaytak — APPROVED** [review]

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs [hunk 5]
```diff
@@ -57,11 +57,24 @@ public static async Task<string> GetMSIEnvironmentVariablesAsync(string uri)
...
+        /// Returns the AAD cloud user idlab1@msidlab4.onmicrosoft.com
+        /// </summary>
+        /// <returns></returns>
         public static Task<LabResponse> GetDefaultUserAsync()
         {
             return GetLabUserDataAsync(UserQuery.PublicAadUserQuery);
         }
 
+        /// <summary>
+        /// Returns the AAD cloud user idlab2@msidlab4.onmicrosoft.com
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs:70]: The user referenced in `PublicAadUser2Query` is actually idlab@msidlab4.onmicrosoft.com

File: tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs [hunk 6]
```diff
@@ -57,11 +57,24 @@ public static async Task<string> GetMSIEnvironmentVariablesAsync(string uri)
             return result;
         }
 
+        /// <summary>
+        /// Returns the AAD cloud user idlab1@msidlab4.onmicrosoft.com
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.LabInfrastructure/LabUserHelper.cs:61]: This one is using "idlab1@msidlab4.onmicrosoft.com"

File: tests/Microsoft.Identity.Test.LabInfrastructure/UserQueryParameters.cs [hunk 7]
```diff
@@ -20,10 +20,19 @@ public class UserQuery
...
+        /// Ask for a specific user from the lab. No other parameters will be considered.
+        /// </summary>
+        public string Upn { get; set; }
+
         public static UserQuery PublicAadUserQuery => new UserQuery()
-            {
-                UserType = LabInfrastructure.UserType.Cloud,
-                AzureEnvironment = LabInfrastructure.AzureEnvironment.azurecloud
+        {
+            Upn = "idlab1@msidlab4.onmicrosoft.com"
```
Comment (trwalke) [inline tests/Microsoft.Identity.Test.LabInfrastructure/UserQueryParameters.cs:30]: are these 2 upns swapped around?

**Comment by trwalke** [comment]
2 minor comments, LGTM otherwise

**Review by trwalke — APPROVED** [review]

**Pull request merged**